--- gst-plugins-bad-1.8.1/ext/qt/Makefile.am.0002~	2016-07-05 22:38:19.778710786 +0200
+++ gst-plugins-bad-1.8.1/ext/qt/Makefile.am	2016-07-06 00:40:17.877765478 +0200
@@ -13,6 +13,8 @@ moc_generated = \
 BUILT_SOURCES = $(moc_generated)
 CLEANFILES = $(moc_generated)
 
+AM_CXXFLAGS = -std=gnu++14
+
 nodist_libgstqtsink_la_SOURCES = $(BUILT_SOURCES)
 
 libgstqtsink_la_SOURCES = \
@@ -30,7 +32,8 @@ libgstqtsink_la_CXXFLAGS = \
 	$(GST_BASE_CFLAGS) \
 	$(GST_PLUGINS_BASE_CFLAGS) \
 	$(QT_CFLAGS) \
-	$(GL_CFLAGS)
+	$(GL_CFLAGS) \
+	$(AM_CXXFLAGS)
 
 libgstqtsink_la_LIBADD = \
 	$(GST_BASE_LIBS) \
