diff -uraN gst-plugins-bad-1.16.2/gst-libs/gst/opencv/meson.build omv-gst-plugins-bad-1.16.2/gst-libs/gst/opencv/meson.build
--- gst-plugins-bad-1.16.2/gst-libs/gst/opencv/meson.build	2019-09-03 02:03:36.000000000 +0200
+++ omv-gst-plugins-bad-1.16.2/gst-libs/gst/opencv/meson.build	2020-06-07 14:41:37.786433330 +0200
@@ -11,7 +11,7 @@
 
 opencv_dep = dependency('opencv', version : '>= 3.0.0', required : false)
 if not opencv_dep.found()
-  opencv_dep = dependency('opencv4', version : '>= 4.0.0', required : false)
+  opencv_dep = dependency('opencv4', version : '>= 4.3.0', required : false)
 endif
 if opencv_dep.found()
   gstopencv = library('gstopencv-' + api_version,
