diff -p -up ./ext/timidity/gstwildmidi.c~ ./ext/timidity/gstwildmidi.c
--- ./ext/wildmidi/gstwildmididec.c~	2008-03-13 15:25:20.000000000 +0100
+++ ./ext/wildmidi/gstwildmididec.c	2008-05-16 16:31:45.000000000 +0200
@@ -59,7 +59,7 @@
 #define WILDMIDI_NUM_CHANNELS 2
 
 #ifndef WILDMIDI_CFG
-#define WILDMIDI_CFG "/etc/timidity.cfg"
+#define WILDMIDI_CFG "/etc/timidity/timidity.cfg"
 #endif
 
 #define DEFAULT_LOG_VOLUME_SCALE     TRUE
