diff -uraN gst-plugins-bad-1.16.2/configure.ac omv-gst-plugins-bad-1.16.2/configure.ac
--- gst-plugins-bad-1.16.2/configure.ac	2019-12-03 12:08:05.000000000 +0100
+++ omv-gst-plugins-bad-1.16.2/configure.ac	2020-06-07 14:55:02.856713153 +0200
@@ -1845,7 +1845,7 @@
       HAVE_OPENCV="yes"
     fi
   ], [
-    PKG_CHECK_MODULES([OPENCV], [opencv4 >= 4.0.0 opencv4 < 4.2.0] , [
+    PKG_CHECK_MODULES([OPENCV], [opencv4 >= 4.0.0 opencv4 < 4.5.0] , [
     AC_PROG_CXX
     AC_LANG([C++])
     OLD_CPPFLAGS=$CPPFLAGS
diff -uraN gst-plugins-bad-1.16.2/ext/opencv/meson.build omv-gst-plugins-bad-1.16.2/ext/opencv/meson.build
--- gst-plugins-bad-1.16.2/ext/opencv/meson.build	2019-09-03 02:03:36.000000000 +0200
+++ omv-gst-plugins-bad-1.16.2/ext/opencv/meson.build	2020-06-07 15:10:56.626691351 +0200
@@ -65,7 +65,7 @@
     endif
   endforeach
 else
-  opencv_dep = dependency('opencv4', version : ['>= 4.0.0', '< 4.2.0'], required : false)
+  opencv_dep = dependency('opencv4', version : ['>= 4.0.0', '< 4.5.0'], required : false)
   opencv_found = opencv_dep.found()
   if opencv_found
     foreach h : libopencv4_headers
diff -uraN gst-plugins-bad-1.16.2/gst-libs/gst/opencv/meson.build omv-gst-plugins-bad-1.16.2/gst-libs/gst/opencv/meson.build
--- gst-plugins-bad-1.16.2/gst-libs/gst/opencv/meson.build	2019-09-03 02:03:36.000000000 +0200
+++ omv-gst-plugins-bad-1.16.2/gst-libs/gst/opencv/meson.build	2020-06-07 14:41:37.786433330 +0200
@@ -11,7 +11,7 @@
 
 opencv_dep = dependency('opencv', version : '>= 3.0.0', required : false)
 if not opencv_dep.found()
-  opencv_dep = dependency('opencv4', version : '>= 4.0.0', required : false)
+  opencv_dep = dependency('opencv4', version : '>= 4.3.0', required : false)
 endif
 if opencv_dep.found()
   gstopencv = library('gstopencv-' + api_version,
