diff -Naur gst-plugins-bad-1.12.0/ext/wildmidi/gstwildmidi.c gst-plugins-bad-1.12.0.tpg/ext/wildmidi/gstwildmidi.c
--- gst-plugins-bad-1.12.0/ext/wildmidi/gstwildmidi.c	2017-04-13 11:10:59.020189152 +0000
+++ gst-plugins-bad-1.12.0.tpg/ext/wildmidi/gstwildmidi.c	2017-05-16 19:48:34.338932993 +0000
@@ -55,7 +55,7 @@
 #include "gstwildmidi.h"
 
 #ifndef WILDMIDI_CFG
-#define WILDMIDI_CFG "/etc/timidity.cfg"
+#define WILDMIDI_CFG "/etc/timidity/timidity.cfg"
 #endif
 
 GST_DEBUG_CATEGORY_STATIC (gst_wildmidi_debug);
